parser.y

```diff
@@ -2036,6 +2040,36 @@ IndexColNameList:
 
 
 
+/**************************************AlterDatabaseStmt***************************************
+ * See https://dev.mysql.com/doc/refman/5.7/en/alter-database.html
+ * 'ALTER DATABASE ... UPGRADE DATA DIRECTORY NAME' is not supported yet.
+ *
+ *  ALTER {DATABASE | SCHEMA} [db_name]
+ *   alter_specification ...
+ *
+ *  alter_specification:
+ *   [DEFAULT] CHARACTER SET [=] charset_name
+ * | [DEFAULT] COLLATE [=] collation_name
+ *******************************************************************************************/
+ AlterDatabaseStmt:
+	"ALTER" DatabaseSym DBNameOpt DatabaseOptionListOpt
```

Comment:
I think it's better to split into two rules rather than passing an `[]interface{}`:

```
    "ALTER" DatabaseSym DBName DatabaseOptionList
|   "ALTER" DatabaseSym DatabaseOptionList
```

BTW why use a `DatabaseOptionListOpt` and then check `len(opts) == 0` instead of using `DatabaseOptionList`?